pyrokinetics/pyroscan.py

```diff
@@ -253,6 +253,12 @@ def load_gk_output(self):
 
         # xarray DataSet to store data
         ds = xr.Dataset(self.parameter_dict)
+        if self.base_pyro.gk_code == "TGLF":
+            nmode = self.base_pyro.gk_input.data.get("nmodes", 2)
+            nmode_coords = {"nmode": list(range(1, 1 + nmode))}
+            ds = ds.assign_coords(nmode_coords)
+        else:
+            nmode = np.nan
```

Comment:
TGLF is an eigensolver and you can request any number of modes for it to find. However, it might not find any modes and in that scenario the outputted data changes. 
 
Mostly we handle initial value solvers so it could be good to distinguish between the two. Add a property like 
`pyro.gk_code.solver_type = 'initial_value'`

Probably best for a separate PR

Reply:
Raised #93